docs/apache-airflow/core-concepts/dags.rst

```diff
@@ -484,6 +484,125 @@ You can also combine this with the :ref:`concepts:depends-on-past` functionality
     .. image:: /img/branch_with_trigger.png
 
 
+.. _concepts:setup-and-teardown:
+
+Setup and Teardown
+~~~~~~~~~~~~~~~~~~
+
+In data workflows it's common to create resources (such as a compute resource), use it to do some work, and then tear it down. Airflow provides setup and teardown tasks to support this need.
+
+Key features of setup and teardown tasks:
+
+  * If you clear a task, its setups and teardowns will be cleared.
+  * By default, teardown tasks are ignored for the purpose of evaluating dag run state.
+  * A teardown task will run if it's setup was successful, even if its work tasks failed.
+  * Teardown tasks are ignored when setting dependencies against task groups.
+  * A setup task must always have a teardown and vice versa. You may use EmptyOperator as a setup or teardown.
+
+Basic usage
+"""""""""""
+
+Suppose you have a dag that creates a cluster, runs a query, and deletes the cluster. Without using setup and teardown tasks you might set these relationships:
+
+.. code-block:: python
+
+  create_cluster >> run_query >> delete_cluster
+
+We can use the ``as_teardown`` let airflow know that ``create_cluster`` is a setup task and ``delete_cluster`` is its teardown:
+
+.. code-block:: python
+
+  create_cluster >> run_query >> delete_cluster.as_teardown(setups=create_cluster)
+
+Observations:
+
+  * If you clear ``run_query`` to run it again, then both ``create_cluster`` and ``delete_cluster`` will be cleared.
+  * If ``run_query`` fails, then ``delete_cluster`` will still run.
+  * The success of the dag run will depend on the success of ``run_query``.
```

Comment:
Would it be accurate here to say "the success of the dag run will depend *only* on the success of `run_query`"? If so, I think that would be a good add to drive that point.

Related, what is the state of the DAG if the setup fails? Does the non-setup/teardown task get marked as skipped in that case?

Reply:
> Would it be accurate here to say "the success of the dag run will depend only on the success of run_query"? If so, I think that would be a good add to drive that point.

Yeah I think that's a good suggestion.

> Related, what is the state of the DAG if the setup fails? Does the non-setup/teardown task get marked as skipped in that case?

If the setup fails, then everything proceeds according to trigger rules.  The work task would be `failed` unless user modifies the trigger rules to do something different.

Reply:
> If the setup fails, then everything proceeds according to trigger rules. The work task would be failed unless user modifies the trigger rules to do something different.

Makes sense, I think that would be a good bullet to add :)